cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -2006,10 +2006,11 @@ class ArrayRanker : public TypeVisitor {
       }
 
       case RankOptions::Min: {
-        T curr_value, prev_value;
+        T curr_value, prev_value{};
+        rank = 0;
 
         if (null_placement_ == NullPlacement::AtStart) {
-          rank = 1;
+          rank++;
```

Comment:
This seems harmless but why this change?

Reply:
Just for consistency for other similar code blocks above.